c/bitvector/sum02_true-unreach-call.c

```diff
@@ -8,9 +9,16 @@ void __VERIFIER_assert(int cond) {
 }
 extern unsigned int __VERIFIER_nondet_uint();
 int main() { 
-  unsigned int i, n=__VERIFIER_nondet_uint(), sn=0;
+  unsigned int i, n=__VERIFIER_nondet_uint();
+  unsigned long long sn=0;
+  // n needs to be less than 2^32 such that the multiplication part
+  // of the Gauss sum does not exceed the range of unsigned long long
+  __VERIFIER_assume(n < 4294967296U); 
   for(i=0; i<=n; i++) {
     sn = sn + i;
   }
-  __VERIFIER_assert(sn==(n*(n+1))/2 || sn == 0);
+  // Compute Gauss sum without overflow
+  unsigned long long nl = n;
+  unsigned long long gauss = (nl*(nl+1U))/2U;
+  __VERIFIER_assert(sn==gauss || sn == 0);
```

Comment:
`sn==0` might be superfluous now. It does not influence the error, thus we keep it.